spec/interactors/process_hubee_notification_spec.rb

```diff
@@ -0,0 +1,89 @@
+require "rails_helper"
+
+RSpec.describe ProcessHubEENotification, type: :interactor do
+  subject(:interactor) { described_class.call(notification: notification_hash) }
+
+  let(:notification_hash) do
+    {
+      "caseId" => "1234",
+      "eventId" => "5678",
+      "processCode" => "FormulaireQF",
+      "id" => "abcd",
+    }
+  end
+
+  describe ".call" do
+    let(:event_payload) do
+      {
+        "actionType" => "STATUS_UPDATE",
+        "attachments" => [],
+        "author" => "John Doe",
+        "caseCurrentStatus" => "SENT",
+        "caseNewStatus" => status,
+        "comment" => "",
+        "createDateTime" => "2024-07-04T08:54:15.438+00:00",
+        "id" => "905055ea-ed37-4556-9db6-97ba89fcb91f",
+        "message" => "",
+        "notification" => true,
+        "partnerAttributes" => nil,
+        "partnerInfo" => {
+          "editorName" => "SOMEEDITOR",
+          "applicationName" => "Portail",
+          "softwareVersion" => "2.2.2",
+        },
+        "status" => "SENT",
+        "transmitter" => {
+          "type" => "SI",
+          "branchCode" => "04107",
+          "companyRegister" => "21040107100019",
+        },
+        "updateDateTime" => nil,
+      }
+    end
+    let(:session) { instance_double(HubEE::Api) }
+    let(:status) { "SI_RECEIVED" }
+
+    before do
+      allow(HubEE::Api).to receive(:session).and_return(session)
+      allow(session).to receive(:event).and_return(event_payload)
+      allow(session).to receive(:update_event)
+      allow(session).to receive(:delete_notification)
+    end
+
+    it "deletes the notification" do
+      expect(session).to receive(:delete_notification)
+      interactor
+    end
+
+    context "when the notification is for FormulaireQF" do
+      context "when the notification is for a new folder" do
+        context "when the event is valid, sent, and a status update" do
```

Comment:
Tu ne fais souvent que le happy path dans tes tests, mais du coup ton coverage est très limité. Il faut penser à tester aussi les autres cas. Sinon ta spec est incomplète.

Reply:
Ca permettrait aussi d'expliciter les différents use cases de la suite de conditions de `ProcessHubEENotification#call` dans les contextes detes tests.

Par exemple :

```ruby
context "when it's a notification of a new folder" do
  let(:event_id) { nil }
  
  it "doesn't process it" do
     ...
  end
  
  it "deletes the notification" do
     ...
  end
end
```

Comme ça, en lisant les tests on comprend mieux l'implémentation de 

```ruby
    return if notification.new_folder?
```

et 

```ruby
    def new_folder?
      event_id.blank?
    end
```

Reply:
Oui c'est vrai mais du coup il ne se passe rien dans ces cas donc c'est pas génial à tester :/

Ou alors il faut qu'on se dise qu'on renvoie une info dans le contexte pour indiquer ce qu'il s'est passé (why not, faut se mettre d'accord sur quoi du coup)